gma/src/include/firebase/gma/internal/native_ad.h

```diff
@@ -57,12 +65,48 @@ class NativeAd {
   /// LoadAd.
   Future<AdResult> LoadAdLastResult() const;
 
+  /// Returns the associated icon asset of the native ad.
+  const NativeAdImage icon() const;
+
+  /// Returns the associated image assets of the native ad.
+  const std::vector<NativeAdImage>& images() const;
```

Comment:
Will you change this by the vector directly or keep it as is?

Reply:
Better to keep it as is. We do maintain a cached copy internally and not generate it on the fly during every function call.